### ISSUE TITLE
Should there be a way to get int64 epochseconds from a datetime?

### ISSUE THREAD

**Issue by brianbegy** [edgedb/edgedb#8133]
- EdgeDB Version: 5.6+95614e9 (cloud)
- EdgeDB CLI Version: 5.5.1+cc9315a
- OS Version: (cloud), client is Mac14.  

Steps to Reproduce:

1.  Run this query:
```
Select  {
  no_cast_returns_float:=datetime_get(datetime_current(), 'epochseconds') * 1000,
  cast_epoch_result_returns_int64:= <int64>datetime_get(datetime_current(), 'epochseconds') * <int64>1000,
  cast_multiplier_returns_float:= datetime_get(datetime_current(), 'epochseconds') * <int64>1000,
  post_cast_returns_int64:=<int64>(datetime_get(datetime_current(), 'epochseconds') * <int64>1000),
}
```

2. Observe that some results are floats.  

This behavior is pretty unexpected and is a trap for the unwary.  

The typescript client does not prevent this action: 
```    
e.op(e.datetime_get(stream.createdAt, 'epochseconds'), '*', 1000)
```

returns a float, although casting after the fact does of course work:

```
  created: e.cast(
    e.int64,
    e.op(e.datetime_get(myentity.createdAt, 'epochseconds'), '*', 1000)
  ),
```

**Comment by scotttrinh** [comment]
> This behavior is pretty unexpected and is a trap for the unwary.

What makes you think that `datetime_get(datetime, 'epochseconds')` should ever return an int64? [It is supposed to return a `float64` according to the reference docs](https://docs.edgedb.com/database/stdlib/datetime#function::std::datetime_get), so the behavior you outlined (where it returns a float unless you cast it) seems like the most reasonable behavior?

**Comment by brianbegy** [comment]
> > This behavior is pretty unexpected and is a trap for the unwary.
> 
> What makes you think that `datetime_get(datetime, 'epochseconds')` should ever return an int64? [It is supposed to return a `float64` according to the reference docs](https://docs.edgedb.com/database/stdlib/datetime#function::std::datetime_get), so the behavior you outlined (where it returns a float unless you cast it) seems like the most reasonable behavior?

With int64 available, it never occurred to me that anyone would choose to represent the number of seconds since 1970 as anything but an int64.  

But "we decided it should be this way" is an answer.  Feel free to close as "works as designed."

**Comment by scotttrinh** [comment]
> With int64 available, it never occurred to me that anyone would choose to represent the number of seconds since 1970 as anything but an int64.

I believe this is an inheritance from Postgres' `date_part` (which returns `DOUBLE PRECISION`) which is what we map our own `datetime_get` to. Choosing int64 and switching to `extract`(which returns `NUMERIC`, meaning a decimal -> int64 conversion I think?) seems like something to consider, but I'll defer to the compiler team on that.

**Comment by msullivan** [comment]
Yeah, this is a pretty weird choice of representation by `postgres` and probably one that we should have fixed instead of faithfully wrapping.

It looks like the thing driving it is a desire for `seconds` and `milliseconds` to be able to return fractional results, which forces everything else to match. I'm skeptical of that plan.

It's probably not worth adding a second way to extract parts of a datetime to fix this, though?

The good news is that float64 has 53 bits of integer precision, so we're in no danger of precision loss.
